### PR TITLE
Added Insertion Sort [Python]

### LINKED ISSUE #84: Insertion Sort [Python]
https://en.wikipedia.org/wiki/Insertion_sort

### PR DESCRIPTION
**Pull request by rajeeviiit**
fixed #84 

#### By submitting this pull request I confirm I've read and complied with the below requirements.

- [x] I have read the [Contribution guidelines](https://github.com/iiitv/algos/blob/master/CONTRIBUTING.md) and I am confident that my PR reflects them.
- [x] I have followed the [coding guidelines](https://github.com/iiitv/algos/blob/master/CONTRIBUTING.md#cs) for this project.
- [x] My code follows the [skeleton code structure](https://github.com/iiitv/algos/blob/master/CONTRIBUTING.md#sample).
- [x] This pull request has a descriptive title. For example, `Added {Algorithm/DS name} [{Language}]`, not `Update README.md` or `Added new code`.

### CODE REVIEW

File: insertion_sort/insertion_sort.py [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+     while position>0 and alist[position-1]>currentvalue:
+         alist[position]=alist[position-1]
+         position = position-1
+     alist[position]=currentvalue
+
+def main():
+    alist = [54,26,93,17,77,31,44,55,20]
+    insertionSort(alist)
+    print(alist)
+
```
Comment (jsroyal) [inline insertion_sort/insertion_sort.py]: please remove space

File: insertion_sort/insertion_sort.py [hunk 2]
```diff
@@ -0,0 +1,22 @@
+def insertionSort(alist):
+   """
+    Performs a insertion Sort
+    :param alist: Iterable of elements   
+    """
+   for index in range(1,len(alist)):
+     currentvalue = alist[index]
+     position = index
+
```
Comment (jsroyal) [inline insertion_sort/insertion_sort.py]: also remove space

File: insertion_sort/insertion_sort.py [hunk 3]
```diff
@@ -0,0 +1,22 @@
+def insertionSort(alist):
+   """
+    Performs a insertion Sort
+    :param alist: Iterable of elements
```
Comment (aviaryan) [inline insertion_sort/insertion_sort.py]: The "Performs a inser.." text has extra whitespace before it..
It is not inline with the `"""` . 
Please fix that.

File: insertion_sort/insertion_sort.py [hunk 4]
```diff
@@ -0,0 +1,20 @@
+def insertionSort(alist):
+   """
+    Performs a insertion Sort
+    :param alist: Iterable of elements   
+    """
+   for index in range(1,len(alist)):
+     currentvalue = alist[index]
+     position = index    
+     while position>0 and alist[position-1]>currentvalue:
+         alist[position]=alist[position-1]
```
Comment (aviaryan) [inline insertion_sort/insertion_sort.py]: Use a consistent style.
You are assigning as `var=value` whereas the best way is `var = value` which you have already used at majority of other places.

File: insertion_sort/insertion_sort.py [hunk 5]
```diff
@@ -0,0 +1,20 @@
...
+         position = position-1
+     alist[position]=currentvalue
+
+def main():
+    alist = [54,26,93,17,77,31,44,55,20]
```
Comment (aviaryan) [inline insertion_sort/insertion_sort.py]: It's generally preferred that the comma is proceeded by a space. 
Example
```py
s = [2, 3, 4]
z = func(2, 3, "string")
print a, b
```

File: insertion_sort/insertion_sort.py (hunk 3 above)
Comment (rajeeviiit) [inline insertion_sort/insertion_sort.py]: whitespace removed

File: insertion_sort/insertion_sort.py (hunk 4 above)
Comment (rajeeviiit) [inline insertion_sort/insertion_sort.py]: style fixed

**Comment by aviaryan** [comment]
@rajeeviiit Some codacy issues still remain.

**Comment by rajeeviiit** [comment]
@aviaryan 

what kind of check is codeclimate getting 1 issue . code is correct what to do next.

**Comment by aviaryan** [comment]
@rajeeviiit See the Files tab in this PR. In addition to the code, you have committed a lot of additional files. This PR can't be merged right now. 
Codeclimate actually detected an issue in one of those accidentally committed files. You can click on Details near the codeclimate check status to see what are the problems btw. 

So you will now have to remove the extra files before this PR can be merged. Hope this helps.

File: insertion_sort/insertion_sort.py [hunk 6]
```diff
@@ -0,0 +1,22 @@
+def insertionSort(alist):
+    """
+    Performs a insertion sort
+    :type alist: object
+    """
+    for index in range(1,len(alist)):
```
Comment (singhpratyush) [inline insertion_sort/insertion_sort.py]: Please put space after comma.

File: insertion_sort/insertion_sort.py [hunk 7]
```diff
@@ -0,0 +1,22 @@
...
+        currentvalue = alist[index]
+        position = index
+    while position>0 and alist[position-1]>currentvalue:
```
Comment (singhpratyush) [inline insertion_sort/insertion_sort.py]: Give space around comparisons, `position > 0 and alist[position-1] > currentvalue`.

File: insertion_sort/insertion_sort.py [hunk 8]
```diff
@@ -0,0 +1,22 @@
+def insertionSort(alist):
+    """
+    Performs a insertion sort
+    :type alist: object
+    """
+    for index in range(1,len(alist)):
+        currentvalue = alist[index]
```
Comment (singhpratyush) [inline insertion_sort/insertion_sort.py]: Make it `current_value`. It gets easier to understand.

**Comment by rajeeviiit** [comment]
@aviaryan 
Sorry, It was just added other files by my IDE.Now, have review please.

File: insertion_sort/insertion_sort.py [hunk 9]
```diff
@@ -0,0 +1,22 @@
...
+        position = index
+    while position>0 and alist[position-1]>currentvalue:
+         alist[position] = alist[position-1]
```
Comment (Monal5031) [inline insertion_sort/insertion_sort.py]: You seem to have indentation errors, please check it once using pep8 (linting tool)

Comment (rajeeviiit) [inline insertion_sort/insertion_sort.py]: Nope! Can you check now?

File: insertion_sort/insertion_sort.py [hunk 10]
```diff
@@ -0,0 +1,22 @@
+def insertionSort(alist):
+    """
+    Performs a insertion sort
+    :type alist: object
+    """
+    for index in range(1, len(alist)):
+        current_value = alist[index]
+        position = index
+    while position > 0 and alist[position-1] > current_value:
+         alist[position] = alist[position-1]
```
Comment (singhpratyush) [inline insertion_sort/insertion_sort.py]: Please use exactly 4 spaces for one indent level.

Comment (rajeeviiit) [inline insertion_sort/insertion_sort.py]: done

**Pull request merged**